sql/src/main/java/io/crate/expression/scalar/string/TrimFunction.java

```diff
@@ -0,0 +1,243 @@
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Chars;
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.sql.tree.TrimMode;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.HashSet;
+import java.util.List;
+
+
+public final class TrimFunction {
+
+    private static final String NAME = "trim";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME, new BaseFunctionResolver(FuncParams
+            .builder(Param.STRING)
+            .withVarArgs(Param.STRING).limitVarArgOccurrences(2)
+            .build()) {
+
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> datatypes) {
+                if (datatypes.size() == 1) {
+                    return new OneCharTrimFunction(
+                        new FunctionInfo(
+                            new FunctionIdent(NAME, ImmutableList.of(datatypes.get(0))),
+                            datatypes.get(0)),
+                        ' '
+                    );
+                } else if (datatypes.size() == 2) {
+                    return new TwoArgsTrimFunction(
+                        new FunctionInfo(new FunctionIdent(NAME, datatypes), datatypes.get(0))
+                    );
+                } else {
+                    return new ThreeArgsTrimFunction(
+                        new FunctionInfo(new FunctionIdent(NAME, datatypes), datatypes.get(0))
+                    );
+                }
+            }
+        });
+    }
+
+    private abstract static class TrimBaseFunction extends Scalar<String, String> {
```

Comment:
I'd remove the base class - `getUniqueChars` isn't used in all sub-classes and could also be a static function. Having the class hierarchy for a simple `FunctionInfo` property is imho not worth it.